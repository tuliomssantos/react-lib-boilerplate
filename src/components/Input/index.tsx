import React, { forwardRef } from 'react'

import clsx from 'clsx'
import styles from './styles.module.scss'

export interface InputProps extends React.ComponentProps<'input'> {
  backgroundColor?: string
  variant?: 'outlined' | 'filled' | 'standard'
}

const Input = forwardRef<HTMLInputElement, InputProps>(
  ({ backgroundColor = '', className, variant = 'standard', ...rest }, ref) => {
    return (
      <input
        {...rest}
        ref={ref}
        className={clsx([styles.input, styles[`input__${variant}`], className])}
        style={{
          ...rest.style,
          backgroundColor,
        }}
      />
    )
  }
)

Input.displayName = 'Input'

export default Input
