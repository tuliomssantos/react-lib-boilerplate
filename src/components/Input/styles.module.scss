.input {
  width: 100%;

  &__standard {
    background: rgba(255, 255, 255, 0.09);
  }

  &__outlined {
    border: 1px solid gray;
  }

  &__filled {
    border-bottom: 1px solid gray;
  }

}
