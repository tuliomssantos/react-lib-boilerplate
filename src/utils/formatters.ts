export const upperCase = (word: string) => word.toUpperCase()
