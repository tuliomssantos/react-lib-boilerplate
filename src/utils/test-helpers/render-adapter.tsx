import React from 'react'

import '@testing-library/jest-dom'

import { render, RenderResult } from '@testing-library/react'

global.React = React

export const renderAdapter = (ui: React.ReactElement): RenderResult => {
  return render(ui)
}
