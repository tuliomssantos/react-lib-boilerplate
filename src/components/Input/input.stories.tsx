import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import Input from '.'

type ComponentType = typeof Input

export default {
  title: 'Components/Input',
  component: Input,
  argTypes: {
    variant: {
      control: { type: 'inline-radio' },
    },
  },
} as ComponentMeta<ComponentType>

// Create a master template for mapping args to render the Input component
const Template: ComponentStory<ComponentType> = args => (
  <Input {...args} style={{ maxWidth: 200 }} />
)

// Reuse that template for creating different stories
export const Standard = Template.bind({})
Standard.args = { placeholder: 'Standard', variant: 'standard' }

export const Filled = Template.bind({})
Filled.args = { placeholder: 'Filled', variant: 'filled' }

export const Outlined = Template.bind({})
Outlined.args = { placeholder: 'Outlined', variant: 'outlined' }
