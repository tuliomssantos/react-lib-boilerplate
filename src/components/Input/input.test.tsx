import React from 'react'

import { renderAdapter } from '../../utils/test-helpers'

import Input from '.'

describe('<Input />', () => {
  test('should render correctly', async () => {
    const { container } = renderAdapter(<Input placeholder="rendered" />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
