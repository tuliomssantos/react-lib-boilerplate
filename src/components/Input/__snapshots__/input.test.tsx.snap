// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly 1`] = `
<input
  class="input input__standard"
  placeholder="rendered"
/>
`;
